### PR TITLE
[5.6] Add JPEG support to FileFactory::image()

### PR DESCRIPTION
**Pull request by SimonStaudenmeir**
Currently, when generating dummy images for testing, `FileFactory::image()` always creates a PNG:

```php
UploadedFile::fake()->image('avatar.png') // Creates PNG file.
UploadedFile::fake()->image('avatar.jpg') // Also creates PNG file.
```

This PR checks the file extension and creates JPEG images when the filename ends with `.jpeg` or `.jpg`.

The tests use `getimagesize()` instead of `$image->getMimeType()`. `getMimeType()` derives the mime type from the filename, while `getimagesize()` examines the actual file.

### CODE REVIEW

File: src/Illuminate/Http/Testing/FileFactory.php [hunk 1]
```diff
@@ -28,7 +30,9 @@ public function create($name, $kilobytes = 0)
      */
     public function image($name, $width = 10, $height = 10)
     {
-        return new File($name, $this->generateImage($width, $height));
+        $type = Str::endsWith($name, ['.jpg', '.jpeg']) ? 'jpeg' : 'png';
```
Comment (tzsk) [inline src/Illuminate/Http/Testing/FileFactory.php]: I would use `strtolower($name)` because in some cases `image.JPG` might also come.

**Comment by crynobone** [comment]
screen shot 2018-07-18 at 6 45 15 pm

This could cause a tests failure if your testing environment is not configure `gd` with `jpeg` support. See <https://stackoverflow.com/questions/19028793/fatal-error-call-to-undefined-function-imagejpeg-on-heroku-php-fix/34764738#34764738>

**Comment by pierrocknroll** [comment]
Same error here... :(

**Comment by mvanduijker** [comment]
If you have failing tests, test with a png image by making sure it has .png file extension.

**Pull request merged**